File: src/model/offre.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package model;

/**
 *
 * @author LENOVO
 */


/**
 *
 * @author LENOVO
 */
public class offre {

  
   private int id_offre;
   private String text_offre;
   private String  date_offre;
   private String type_offre;
   private String img_offre;
    public offre(){ }

    public offre(int id_offre, String text_offre, String date_offre, String type_offre) {
        this.id_offre = id_offre;
        this.text_offre = text_offre;
        this.date_offre = date_offre;
        this.type_offre = type_offre;
    }

    public offre(String text_offre, String date_offre, String type_offre) {
        this.text_offre = text_offre;
        this.date_offre = date_offre;
        this.type_offre = type_offre;
    }

    public offre(String text_offre, String date_offre, String type_offre, String img_offre) {
        this.text_offre = text_offre;
        this.date_offre = date_offre;
        this.type_offre = type_offre;
        this.img_offre = img_offre;
    }

    public offre(offre o ) {
        this.id_offre = o.id_offre;
        this.text_offre = o.text_offre;
        this.date_offre = o.date_offre;
        this.type_offre = o.type_offre;
        this.img_offre = o.img_offre;
       
    }

    public offre(int id_offre, String text_offre, String date_offre, String type_offre, String img_offre) {
        this.id_offre = id_offre;
        this.text_offre = text_offre;
        this.date_offre = date_offre;
        this.type_offre = type_offre;
        this.img_offre = img_offre;
    }
    

    public String getImg_offre() {
        return img_offre;
    }

 
  
   
    public int getId_offre() {
        return id_offre;
    }

    public void setId_offre(int id_offre) {
        this.id_offre = id_offre;
    }

    public String getText_offre() {
        return text_offre;
    }

    public void setText_offre(String text_offre) {
        this.text_offre = text_offre;
    }

    public String getDate_offre() {
        return date_offre;
    }

    public void setDate_offre(String date_offre) {
        this.date_offre = date_offre;
    }

    public void setImg_offre(String img_offre) {
        this.img_offre = img_offre;
    }

   

    public String getType_offre() {
        return type_offre;
    }

    public void setType_offre(String type_offre) {
        this.type_offre = type_offre;
    }

    @Override
    public String toString() {
        return "offre{" + "id_offre=" + id_offre + ", text_offre=" + text_offre + ", date_offre=" + date_offre + ", type_offre=" + type_offre + ",img_offre=" + img_offre +'}';
    }

    
            
}

File: src/Entity/Offre.php
<?php

namespace App\Entity;
use App\Controller\uploader;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\Common\Persistence\ManagerRegistry;
use Symfony\Component\HttpFoundation\File\File;
use Symfony\Component\HttpFoundation\File\UploadedFile;
use Vich\UploaderBundle\Mapping\Annotation as Vich;
/**
 * Offre
 *
 * @ORM\Table(name="offre")
 * @ORM\Entity
 * @Vich\Uploadable
 */
class Offre
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_offre", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idOffre;

    /**
     * @var string
     * @Assert\NotBlank(message="description  doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "doit etre >=7 ",
     *      maxMessage = "doit etre <=15" )
     *
     * @ORM\Column(name="text_offre", type="string", length=100, nullable=false)
     */
    private $textOffre;

    /**
     * @var \DateTime
     * @Assert\NotBlank(message="description  doit etre non vide")
     *
     * @ORM\Column(name="date_offre", type="date", nullable=false)
     */
    private $dateOffre;

    /**
     * @var string
     * @Assert\NotBlank(message="description  doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "doit etre >=7 ",
     *      maxMessage = "doit etre <=15" )
     *
     * @ORM\Column(name="type_offre", type="string", length=60, nullable=false)
     */
    private $typeOffre;

    /**
     * @var string
     * @Assert\NotBlank(message="description  doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "doit etre >=7 ",
     *      maxMessage = "doit etre <=15" )
     *
     *
     * @ORM\Column(name="img_offre", type="string", length=100, nullable=false)
     */
    private $imgOffre;
    /**
     * @Vich\UploadableField(mapping="offre_images", fileNameProperty="imgOffre")
     * @var File
     */
    private $imageFile;

    /**
     * @return int
     */
    public function getIdOffre(): int
    {
        return $this->idOffre;
    }

    /**
     * @param int $idOffre
     * @return Offre
     */
    public function setIdOffre(int $idOffre): Offre
    {
        $this->idOffre = $idOffre;
        return $this;
    }

    /**
     * @return string
     */
    public function getTextOffre(): ?string
    {
        return $this->textOffre;
    }

    /**
     * @param string $textOffre
     * @return Offre
     */
    public function setTextOffre(string $textOffre): Offre
    {
        $this->textOffre = $textOffre;
        return $this;
    }

    /**
     * @return \DateTime
     */
    public function getDateOffre(): ?\DateTime
    {
        return $this->dateOffre;
    }

    /**
     * @param \DateTime $dateOffre
     * @return Offre
     */
    public function setDateOffre(\DateTime $dateOffre): Offre
    {
        $this->dateOffre = $dateOffre;
        return $this;
    }

    /**
     * @return string
     */
    public function getTypeOffre(): ?string
    {
        return $this->typeOffre;
    }

    /**
     * @param string $typeOffre
     * @return Offre
     */
    public function setTypeOffre(string $typeOffre): Offre
    {
        $this->typeOffre = $typeOffre;
        return $this;
    }

    /**
     * @return string|null
     */
    public function getImgOffre(): ?string
    {
        return $this->imgOffre;
    }

    /**
     * @param string|null $imgOffre
     * @return $this
     */
    public function setImgOffre(string $imgOffre): self
    {
        $this->imgOffre = $imgOffre;
        return $this;
    }
    /**
     * @return File|null
     */
    public function getImageFilename(): ?File
    {
        return $this->imageFile;
    }
    /**
     * @return File|null $imageFile
     */
    public function setImageFilename(File $imgOffre = null): ?File
    {
        $this->imageFile = $imgOffre;
        return $this;
    }
    public function getImagePath()
    {
        return uploader::PROFILE_IMAGE.'/'.$this->getImageFilename();
    }

}
